Sources/System/FilePath/FilePathSyntax.swift

```diff
@@ -0,0 +1,353 @@
+// Query API
+// @available(...)
+extension FilePath {
+  /// Returns true if this path uniquely identifies the location of
+  /// a file without reference to an additional starting location.
+  ///
+  /// On Unix platforms, absolute paths begin with a `/`. `isAbsolute` is equivalent
+  /// to `root != nil`.
+  ///
+  /// On Windows, absolute paths are fully qualified paths. UNC paths and device paths
+  /// are always absolute. Traditional DOS paths are absolute if they begin with a volume or drive
+  /// followed by a `:` and a separator.
+  ///
+  /// NOTE: This does not perform shell expansion or substitute
+  /// environment variables; paths beginning with `~` are considered relative.
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `/usr/local/bin`
+  ///   * `/tmp/foo.txt`
+  ///   * `/`
+  /// * Windows:
+  ///   * `C:\Users\`
+  ///   * `\\?\UNC\server\share\bar.exe`
+  ///   * `\\server\share\bar.exe`
+  ///
+  /// TODO: We should add API to system to access environment variables
+  /// and perform expansion/substituion, but that's deferred for now. Should
+  /// we mention that?
+  public var isAbsolute: Bool {
+    self.root?._rootIsAbsolute ?? false
+  }
+
+  /// Returns true if this path is not absolute (see absolute)
+  ///
+  /// Examples:
+  /// * Unix:
+  ///   * `~/bar`
+  ///   * `tmp/foo.txt`
+  /// * Windows:
+  ///   * `bar\baz`
+  ///   * `C:Users\`
+  ///   * `\Users`
+  public var isRelative: Bool { !isAbsolute }
+
+  /// A textual representation of the file path using the generic separator (`/`).
+  ///
+  /// On Unix, this is equivalent to `description`
+  ///
+  /// TODO: Should `dump()` use this? Also, should dump escape invalid Unicode?
+  public var genericDescription: String {
```

Comment:
Is this intended only for display purposes?  If so, perhaps it should be called `displayDescription`?  Is there also a way to get a string form that is guaranteed to be reversible, for example by serializing to a JSON and being able to be read back in?

Reply:
A path on Windows can be printed in a platform-preferred representation using backslash, or in a more generic/portable fashion using forwards slashes. `FilePath.description` uses the backslash. We do need a good name for this and I was also considering "portableDescription".

We might make this only available on Windows since its presence is unneeded cognitive overhead for POSIXy users. The counter argument would be that code which wants this representation would need the `#if` in it.

Reply:
Right, I do think that we'd want to avoid `#if` at the callsites where possible.  Maybe a better way for me to phrase the question is "what API would a client use to get a portable description that can be stored in a file and reconstituted on any platform".

Reply:
Resolution: Defer this property from this pitch. Ideally we'd look at RFC-1808 for guidance. Corner cases are application specific. String is not the ideal round trip format due to encoding correction.